org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JavaProjectClasspathChangeAnalyzer.java

```diff
@@ -71,19 +82,71 @@ public boolean isClasspathChangeOnProject(IJavaElementDelta delta) {
 	 * Determines if a given change is a attachment change only
 	 */
 	public boolean isAttachmentChangeOnly(IJavaElementDelta delta) {
-		// there must be at least one child delta
-		if (delta.getAffectedChildren().length == 0) {
-			return false;
+		if (!isJdtCoreGreaterOrEqual(VERSION_3_33_0)) {			
+			if (delta.getAffectedChildren().length == 0) {
+				return true;
+			}
+		} else {
+			// there must be at least one child delta
+			if (delta.getAffectedChildren().length == 0) {
+				return false;
+			}
 		}
 		// all child deltas have to be attachment changes only
 		for (IJavaElementDelta child : delta.getAffectedChildren()) {
-			if (!isAttachmentChangeFlagOnly(child)) {
+			if (!isSourceAttachOrDetachOrJavadocAttributeChangeOnly(child)) {
 				return false;
 			}
 		}
 		return true;
 	}
 	
+	private static Version installedJdtCoreVersion;
+	
+	private static boolean isJdtCoreGreaterOrEqual(Version version) {
+		if (installedJdtCoreVersion == null) {
+			installedJdtCoreVersion = JavaCore.getPlugin().getBundle().getVersion();
+		}
+		return installedJdtCoreVersion.compareTo(version) >= 0;
+	}
+
+	protected boolean isSourceAttachOrDetachOrJavadocAttributeChangeOnly(IJavaElementDelta child) {
+		int flags = child.getFlags();
+		if (flags == 0) {
+			return false;
+		}
+		if ((flags & IJavaElementDelta.F_SOURCEATTACHED) != 0) {
```

Comment:
I think you can work unconditionally with `int flagsWithoutSourceAttachments = flags & ~(F_SOURCE_ATTCHED | F_SOURCE_DETACHED)`